Conceptpower+Spring/src/main/java/edu/asu/conceptpower/servlet/xml/XMLConceptMessage.java

```diff
@@ -73,10 +73,19 @@ public String getEntry(ConceptEntry entry, ConceptType type) {
         List<ChangeEvent> changeEvents = entry.getChangeEvents();
         Collections.sort(changeEvents);
 
-        sb.append("<" + XMLConstants.NAMESPACE_PREFIX + ":" + XMLConstants.CREATOR_ID + ">");
-        sb.append(StringEscapeUtils.escapeXml10(
-                changeEvents.get(0).getUserName() != null ? changeEvents.get(0).getUserName().trim() : ""));
-        sb.append("</" + XMLConstants.NAMESPACE_PREFIX + ":" + XMLConstants.CREATOR_ID + ">");
+        if (changeEvents != null && !changeEvents.isEmpty()) {
+            sb.append("<" + XMLConstants.NAMESPACE_PREFIX + ":" + XMLConstants.CREATOR_ID + ">");
+            sb.append(StringEscapeUtils.escapeXml10(
+                    changeEvents.get(0).getUserName() != null ? changeEvents.get(0).getUserName().trim() : ""));
```

Comment:
making the assumption that at position 0 there is always the first changeevent (used during creation) is fine, but you then have to make sure nobody can change that. This means your get method for change events should never return the original list (otherwise anyone can modify them, change the order, delete objects, etc), and it should provide an add method to add new events that makes sure, events are added at the end of the list. This might seem trivial, but as the system grows people will forgot or not know that you made this basic assumption.